execution/executor/src/lib.rs

```diff
@@ -899,12 +927,6 @@ impl<V: VMExecutor> BlockExecutor for Executor<V> {
             )?;
         }
 
-        // Prune the tree.
-        for block in blocks {
-            for txn_data in block.output().transaction_data() {
-                txn_data.prune_state_tree();
-            }
-        }
```

Comment:
These being removed, the variable `blocks` didn't get dropped at this point, which are locked mutexes, and that's why you are getting a deadlock afterwards.

I think you should do something like the following, but I'm not 100% sure.

```
        for block in blocks {
            block.output().executed_trees().state_tree().prune()
        }
```

Reply:
I think this is still useful here for the logic anyway, if my understanding of what was going on is correct.

Reply:
This seems to be the right fix! Thanks so much for pointing this out!